prescient/engine/egret/engine.py

```diff
@@ -233,33 +260,64 @@ def create_and_solve_lmp(self,
             lmp_sced_instance.data['system']['reserve_shortfall_cost'] = \
                     options.reserve_price_threshold
 
-        self._ptdf_manager.mark_active(lmp_sced_instance)
-        pyo_model = create_KOW_unit_commitment_model(lmp_sced_instance, relaxed=True,
-                                                     ptdf_options = self._ptdf_manager.lmpsced_ptdf_options,
-                                                     PTDF_matrix_dict=self._ptdf_manager.PTDF_matrix_dict)
-
-        self._p._zero_out_costs(pyo_model, self._hours_in_objective)
+        if self._last_sced_pyo_model is None:
+            self._ptdf_manager.mark_active(lmp_sced_instance)
+            pyo_model = create_sced_uc_model(lmp_sced_instance, relaxed=True,
+                                             ptdf_options = self._ptdf_manager.lmpsced_ptdf_options,
+                                             PTDF_matrix_dict=self._ptdf_manager.PTDF_matrix_dict)
+            pyo_solver = self._sced_solver
+            self._p._zero_out_costs(pyo_model, self._hours_in_objective)
```

Comment:
Probably not for this PR, but we should eventually cull names referencing specific units of time from the code base.

Reply:
Generalizing the time step will require lots of changes..

Reply:
Agree - that's a different PR.